### PR TITLE
fix: add form data to submissions earlier on, so we can search on it

### PR DESCRIPTION
**Pull request by jhackett1**
it's ideal to be able to type in the name of a form on the dashboard and see the submissions that match it.

> the filter field on the submissions page seems to work strangely. for instance, if i type “review” it filters out the review 3c forms (meaning, they no longer show) in “just mine” but i see other people’s in “everyone”

this wasn't possible before because we were just querying what came back from the api, which had only the machine-readable `formId`.

- we now use a memoised function to add the form to each submission, before applying searching logic
- we modify components lower down the tree to use the form that already exists, rather than re-searching for it

### CODE REVIEW

File: components/SubmissionsTable/index.spec.tsx [hunk 1]
```diff
@@ -42,6 +43,16 @@ describe('SubmissionsTable', () => {
     expect(screen.getByText('No results to show.'));
   });
 
+  it('lets you search for a submission by form name', async () => {
+    render(<SubmissionsTable submissions={[mockSubmission]} />);
+
+    fireEvent.change(screen.getByRole('searchbox'), {
+      target: { value: 'Sandbox form' },
+    });
```
Comment (Renny4Real) [inline components/SubmissionsTable/index.spec.tsx:51]: Just to check I understand this: when we mock a submission in our tests, this submission is related to the form with an id of `foo` saved [here](https://github.com/LBHackney-IT/lbh-social-care-frontend/blob/f8cb4b2db9a976c96307336b1dfddba1297c5cb4/data/flexibleForms/foo.ts) and that form has the name: `Sandbox form`, which is why we search for it in this test?

File: components/SubmissionsTable/index.tsx [hunk 2]
```diff
@@ -22,20 +23,28 @@ export const SubmissionsTable = ({
...
-          !user?.hasUnrestrictedPermissions &&
-          submission.residents.every((resident) => resident.restricted === 'Y')
-        )
-          return false;
+      submissions
+        // augment each one with its form
+        .map((submission) => ({
+          ...submission,
+          form: forms.find((form) => form.id === submission.formId),
+        }))
```
Comment (Renny4Real) [inline components/SubmissionsTable/index.tsx:31]: Is this somehow pairing a submission with its original form?

**Review by Renny4Real — APPROVED** [review]
Looks good 🎉 . 
Just a couple of questions

File: components/SubmissionsTable/index.tsx (hunk 2 above)
Comment (jhackett1) [reply components/SubmissionsTable/index.tsx:31]: yes - the forms are json files stored in the git repo. the submission stores a `formId` which we can relate back to a particular form, particularly the human-readable name of the form

File: components/SubmissionsTable/index.spec.tsx (hunk 1 above)
Comment (jhackett1) [reply components/SubmissionsTable/index.spec.tsx:51]: exactly. were we to ever change that form this test would break

**Pull request merged**